geopandas/io/tests/test_file.py

```diff
@@ -616,7 +621,6 @@ def test_read_file_geojson_string_path(engine):
         "type": "FeatureCollection",
         "features": [
             {
-                "id": "0",
```

Comment:
Removed the `id` here. Could also have added it to the `expected` dataframe, but given this test is not about the reading of id, removed it to keep it simpler. This should fix most of the failures.